src/App.js

```diff
@@ -1,16 +1,48 @@
 import React from 'react';
 import './App.css';
+import ChatLog from './components/ChatLog';
 import chatMessages from './data/messages.json';
+import { useState } from 'react';
 
 const App = () => {
+  const initialCopy = chatMessages.map((chatentry) => {
+    return { ...chatentry };
+  });
+
+  const [chatData, setChatData] = useState(initialCopy);
+
+  const updateLikes = (chatentryId) => {
+    console.log('updateLikes called');
+    const newChatEntries = [];
+
+    for (const chat of chatData) {
+      if (chat.id !== chatentryId) {
+        newChatEntries.push(chat);
+      } else {
+        const newChat = {
+          ...chat,
+          liked: !chat.liked,
+        };
+        newChatEntries.push(newChat);
+      }
+      setChatData(newChatEntries);
+    }
+  };
+  const countLikes = () => {
```

Comment:
Instead of counting all of likes each time, it would work to keep the number of likes in state. Each time the like button is pressed, that count can go up by one or down by one.